libraries/chain/host_context.cpp

```diff
@@ -62,8 +71,13 @@ uint32_t host_context::execute_sync_call(name call_receiver, uint64_t flags, std
          }
 
          try {
+            // If the call is initiated from an apply_context (an action), it is the first
+            // level of a sync call by the action, set septh to 1.
+            // Otherwise the call is from another sync call, increment the call depth by 1.
+            const uint32_t depth = dynamic_cast<apply_context*>(this) ? 1 : recurse_depth + 1;
```

Comment:
This seems rather ick to me. Can you not pass in the recurse_depth instead.

Reply:
Thanks for this good catch! I was trying to consolidate the depth field but forgot about recurse_depth could be already set by inline actions.

Fixed by https://github.com/AntelopeIO/spring/pull/1285/commits/7d16b88cf172aa95a78344e0d3c17f9aea463426